ansible/roles-infra/infra-ec2-template-generate/templates/cloud_template.j2

```diff
@@ -289,29 +293,31 @@ Resources:
     DependsOn:
       - {{instance['name']}}{{loop.index}}EIP
     Properties:
-      {% if secondary_stack is defined %}
+{%       if not aws_dns_create_public_zone | bool %}
+      HostedZoneName: "{{ aws_dns_zone_root }}"
+{%       elif secondary_stack is defined %}
       HostedZoneName: "{{ aws_dns_zone_public }}"
-      {% else %}
+{%       else %}
       HostedZoneId:
         Ref: DnsZonePublic
-      {% endif %}
+{%       endif %}
```

Comment:
same here

Reply:
also changed here to give `secondary_stack` more weight than `aws_dns_create_public_zone`